packages/http-server/src/server.ts

```diff
@@ -10,63 +11,75 @@ import { IPrismDiagnostic } from '@stoplight/prism-core';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TE from 'fp-ts/lib/TaskEither';
 import * as E from 'fp-ts/lib/Either';
+import { MicriHandler } from 'micri/dist';
+
+function searchParamsToNameValues(searchParams: URLSearchParams): IHttpNameValues {
+  const params = {};
+  for (const key of searchParams.keys()) {
+    const values = searchParams.getAll(key);
+    params[key] = values.length === 1 ? values[0] : values;
+  }
+  return params;
+}
+
+function addressInfoToString(address: AddressInfo | string | null) {
```

Comment:
Given the successive casting, I guess we can just modify the signature to not accept the `string` as a type

```suggestion
function addressInfoToString(address?: AddressInfo) {
```

Even better, I would move the null check outside of this function

Reply:
It is quite weird for me, but the typing says that node's ServerInstance.address can return AddressInfo, string or null. I couldn't make it return something other than AddresssInfo but the typings is the typings..

Reply:
A string is returned if the server is listening on on a pipe or Unix domain socket; for TCP Socket is always going to be `AddressInfo`

Reply:
I would prefer to keep the function clean but this time it will put too much garbage into the place where it is called.

I would need to cast it like that: `addressInfoToString(server.address() as AddressInfo | undefined)` to keep compiler satisfied.

Are you ok with how it looks now or you have other preference?